api/v1alpha3/cluster_types.go

```diff
@@ -138,6 +138,10 @@ type ClusterStatus struct {
 	// Conditions defines current service state of the cluster.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the latest generation observed by the controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration"`
```

Comment:
We should remove `omitempty` in MachineDeployment and MachineSet types to be consistent with these, and make sure e2e and integration tests pass as well